### PR TITLE
FPASF-581 Enabling Towns Fund Round-6 Submission

### PR DESCRIPTION
**Pull request by israr-ulhaq**
https://dluhcdigital.atlassian.net/browse/FPASF-581

This PR is about to Enabling Towns Fund to accept R6 submission.
Updating integration tests for R6 by creating new failure and success spreadsheets.

- [x] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### CODE REVIEW

File: tests/resources/reporting_round.csv [hunk 1]
```diff
@@ -4,10 +4,12 @@ bb265da3-999e-4282-aa2c-0fa85da503a1,2,4a6e9f7d-fc9d-4c12-b1b6-89e784c310e1,2022
 de52f699-f26f-44ff-b7ca-5cc834300500,3,4a6e9f7d-fc9d-4c12-b1b6-89e784c310e1,2022-10-01 00:00:00,2023-03-31 23:59:59,,
 eede818b-8697-4034-a4a2-7c35dff080bb,4,4a6e9f7d-fc9d-4c12-b1b6-89e784c310e1,2023-04-01 00:00:00,2023-09-30 23:59:59,,2023-12-04 23:59:59
 4428f656-5997-42c2-9149-b2303a6f1765,5,4a6e9f7d-fc9d-4c12-b1b6-89e784c310e1,2023-10-01 00:00:00,2024-03-31 23:59:59,,2024-05-28 23:59:59
+4428f654-5997-42c2-9149-b2303a6f1765,6,4a6e9f7d-fc9d-4c12-b1b6-89e784c310e1,2024-04-01 00:00:00,2024-09-30 23:59:59,,
```
Comment (samuelhwilliams) [inline tests/resources/reporting_round.csv:7]: This need to be new random uuid4s, not copies (try `flask db-data seed-ref` to test that it ingests correctly). Also - can we get reporting period start+end populated too?

File: tests/resources/reporting_round.csv [hunk 2]
```diff
@@ -4,10 +4,12 @@ bb265da3-999e-4282-aa2c-0fa85da503a1,2,4a6e9f7d-fc9d-4c12-b1b6-89e784c310e1,2022
...
 eede818b-8697-4034-a4a2-7c35dff080bb,4,4a6e9f7d-fc9d-4c12-b1b6-89e784c310e1,2023-04-01 00:00:00,2023-09-30 23:59:59,,2023-12-04 23:59:59
 4428f656-5997-42c2-9149-b2303a6f1765,5,4a6e9f7d-fc9d-4c12-b1b6-89e784c310e1,2023-10-01 00:00:00,2024-03-31 23:59:59,,2024-05-28 23:59:59
+4428f654-5997-42c2-9149-b2303a6f1765,6,4a6e9f7d-fc9d-4c12-b1b6-89e784c310e1,2024-04-01 00:00:00,2024-09-30 23:59:59,,
 311b7d4b-3545-48fc-bc9a-b2fee38ccd68,1,9fde58b2-8a89-4b2c-af7d-1f968b03c7b9,2019-04-01 00:00:00,2022-03-31 23:59:59,,
 405dc124-06b0-497c-a45a-a0947045fb10,2,9fde58b2-8a89-4b2c-af7d-1f968b03c7b9,2022-04-01 00:00:00,2022-09-30 23:59:59,,
 c9e10f59-419e-4da2-a92b-2f9f9a09923d,3,9fde58b2-8a89-4b2c-af7d-1f968b03c7b9,2022-10-01 00:00:00,2023-03-31 23:59:59,,
 49219b20-3762-4fd8-a591-53508c68b2dd,4,9fde58b2-8a89-4b2c-af7d-1f968b03c7b9,2023-04-01 00:00:00,2023-09-30 23:59:59,,2023-12-04 23:59:59
 0af7ee1e-6c97-4a12-8f4d-59158a9888d4,5,9fde58b2-8a89-4b2c-af7d-1f968b03c7b9,2023-10-01 00:00:00,2024-03-31 23:59:59,,2024-05-28 23:59:59
+4428f345-5997-42c2-9149-b2303a6f1765,6,9fde58b2-8a89-4b2c-af7d-1f968b03c7b9,2024-04-01 00:00:00,2024-09-30 23:59:59,,
```
Comment (samuelhwilliams) [inline tests/resources/reporting_round.csv:13]: This need to be new random uuid4s, not copies (try `flask db-data seed-ref` to test that it ingests correctly). Also - can we get reporting period start+end populated too?

File: data_store/transformation/towns_fund/tf_transform_r3.py [hunk 3]
```diff
@@ -352,7 +352,7 @@ def extract_project_progress(df_data: pd.DataFrame, project_lookup: dict, report
     :return: A new DataFrame containing the extracted project progress rows.
     """
     # if round 4 or 5, ingest two additional columns
-    df_data = df_data.iloc[18:39, 2:15] if reporting_round in (4, 5) else df_data.iloc[18:39, 2:13]
+    df_data = df_data.iloc[18:39, 2:15] if reporting_round >= 4 else df_data.iloc[18:39, 2:13]
```
Comment (samuelhwilliams) [inline data_store/transformation/towns_fund/tf_transform_r3.py:355]: @wjrm500  - thoughts about this or should we be duplicating into a new r6?

File: config/envs/default.py [hunk 4]
```diff
@@ -73,6 +73,7 @@ class DefaultConfig(object):
     DISABLE_LOAD = "DISABLE_LOAD" in os.environ
     ENABLE_TF_R5 = os.getenv("ENABLE_TF_R5", True)
```
Comment (samuelhwilliams) [inline config/envs/default.py:74]: Can we drop this out if we've got ENABLE_TF_R6 now?

File: data_store/transformation/towns_fund/tf_transform_r3.py (hunk 3 above)
Comment (wjrm500) [reply data_store/transformation/towns_fund/tf_transform_r3.py:355]: Have already discussed this in the meeting this afternoon but just to get my thoughts down briefly on paper - reckon it's okay like this, the question of how much code we want to share / duplicate across rounds, and how we handle reporting round transition / deprecation is an interesting one, but I guess my overriding thought here is just to minimise code changes related to TF

File: tests/resources/reporting_round.csv (hunk 1 above)
Comment (wjrm500) [reply tests/resources/reporting_round.csv:7]: As mentioned in the meeting one-character tweaks were made after copying but yeah it's not obvious at all at a glance haha! @israr-ulhaq would you mind looking at this one?

File: tests/resources/reporting_round.csv [hunk 5]
```diff
@@ -4,12 +4,12 @@ bb265da3-999e-4282-aa2c-0fa85da503a1,2,4a6e9f7d-fc9d-4c12-b1b6-89e784c310e1,2022
...
-4428f654-5997-42c2-9149-b2303a6f1765,6,4a6e9f7d-fc9d-4c12-b1b6-89e784c310e1,2024-04-01 00:00:00,2024-09-30 23:59:59,,
+52062ae1-e947-4fc4-9ce1-be5d7887f42f,6,0fe2cf84-6b3c-4dda-96ec-632c76c6dc05,2024-04-01 00:00:00,2024-09-30 23:59:59,2024-10-21 00:00:00,2024-11-24 23:59:59
 311b7d4b-3545-48fc-bc9a-b2fee38ccd68,1,9fde58b2-8a89-4b2c-af7d-1f968b03c7b9,2019-04-01 00:00:00,2022-03-31 23:59:59,,
 405dc124-06b0-497c-a45a-a0947045fb10,2,9fde58b2-8a89-4b2c-af7d-1f968b03c7b9,2022-04-01 00:00:00,2022-09-30 23:59:59,,
 c9e10f59-419e-4da2-a92b-2f9f9a09923d,3,9fde58b2-8a89-4b2c-af7d-1f968b03c7b9,2022-10-01 00:00:00,2023-03-31 23:59:59,,
 49219b20-3762-4fd8-a591-53508c68b2dd,4,9fde58b2-8a89-4b2c-af7d-1f968b03c7b9,2023-04-01 00:00:00,2023-09-30 23:59:59,,2023-12-04 23:59:59
 0af7ee1e-6c97-4a12-8f4d-59158a9888d4,5,9fde58b2-8a89-4b2c-af7d-1f968b03c7b9,2023-10-01 00:00:00,2024-03-31 23:59:59,,2024-05-28 23:59:59
-4428f345-5997-42c2-9149-b2303a6f1765,6,9fde58b2-8a89-4b2c-af7d-1f968b03c7b9,2024-04-01 00:00:00,2024-09-30 23:59:59,,
+ff99f14b-8078-4e5b-99a2-475dc4aca51e,6,d57b28fb-a4f5-465c-ad51-3c7fe55669cb,2024-04-01 00:00:00,2024-09-30 23:59:59,2024-10-21 00:00:00,2024-11-24 23:59:59
```
Comment (samuelhwilliams) [inline tests/resources/reporting_round.csv:13]: Please have another look at and think about these changes - currently you've changed a bit too much so these are broken.

**Comment by samuelhwilliams** [comment]
This need a rebase now please 🙏  Careful with fixing the conflicts - let me know if you want to pair

File: submit/main/fund.py [hunk 6]
```diff
@@ -84,12 +84,12 @@ def get_active_funds(self, roles: list[str]):
 TOWNS_FUND_APP_CONFIG = FundConfig(
     fund_name="Towns Fund",
     user_role="TF_MONITORING_RETURN_SUBMITTER",
-    current_reporting_period="October to March 2024",
-    current_reporting_round=5,
-    current_deadline=datetime.date(day=28, month=5, year=2024),
+    current_reporting_period="April to September 2024",
+    current_reporting_round=6,
+    current_deadline=datetime.date(day=30, month=11, year=2024),
```
Comment (samuelhwilliams) [inline submit/main/fund.py:89]: In the `reporting_round.csv` file below we have the deadline as 24 November 2024 - here we say 30 November 2024.

Which is correct? Can we be consistent?

Comment (israr-ulhaq) [reply submit/main/fund.py:89]: I believe it's 24th November as Craig mentioned it to @wjrm500 ?

**Review by samuelhwilliams — APPROVED** [review]

**Pull request merged**